hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/HRegion.java

```diff
@@ -8265,14 +8265,17 @@ private WALEdit reckonDeltas(Operation op, Mutation mutation, Durability effecti
           break;
         default: throw new UnsupportedOperationException(op.toString());
       }
-      int newCellSize = PrivateCellUtil.estimatedSerializedSizeOf(newCell);
-      if (newCellSize > this.maxCellSize) {
-        String msg = "Cell with size " + newCellSize + " exceeds limit of " + this.maxCellSize
-          + " bytes in region " + this;
-        if (LOG.isDebugEnabled()) {
+      if (this.maxCellSize > 0) {
+        int newCellSize = PrivateCellUtil.estimatedSerializedSizeOf(newCell);
+        if (newCellSize > this.maxCellSize) {
+          String msg = "Cell with size " + newCellSize + " exceeds limit of " + this.maxCellSize
+            + " bytes in region " + this;
           LOG.debug(msg);
+          throw new DoNotRetryIOException(msg);
         }
-        throw new DoNotRetryIOException(msg);
+      } else {
+        LOG.debug("Cell size check is disabled because of maxCellSize is set to {}.",
```

Comment:
Don't need to log this everytime? Only log once is enough when regionserver start.

Reply:
This parameter can be configured to the column family of the table, different tables can have different values.

Reply:
ok. I thought this log is not needed. If log this everytime, there are too many logs.

Reply:
This log may be not very helpful, @wchevreuil  what do you think ?

Reply:
I removed this log,  because there may be too many logs and not very helpful.

Reply:
> I removed this log, because there may be too many logs and not very helpful.

Well, that's why I suggested it to be DEBUG. Logging it only once may not be sufficient in cases where log rolls too quickly.

Reply:
> Well, that's why I suggested it to be DEBUG. Logging it only once may not be sufficient in cases where log rolls too quickly.
We could check it in the regionserver configuration and table desc. It ’s OK to remove this log?

Reply:
Many users and even experienced developers may not be aware of this changed behaviour. When things don't seem to work as expected, and one doesn't have an idea about why, decreasing the log level to gather as much info as possible is a common practice. I know it would be verbose, but that's why we should make it either a DEBUG or a TRACE level message. The message then would give a clear clue for those who didn't even know about the config property or table attribute.